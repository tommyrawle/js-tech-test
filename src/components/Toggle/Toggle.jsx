import React from 'react';
import { connect } from 'react-redux';
import * as actions from '../../redux/actions';
import PropTypes from 'prop-types';
import { ToggleWrapper, ToggleButtonWrapper, ToggleButton } from './Toggle.styles.jsx';

export const Toggle = ({ oddsFormat, setOddsFormat }) => {
  return (
    <ToggleWrapper>
      Odds format
      <ToggleButtonWrapper>
        <ToggleButton
          id="fraction"
          left
          isActive={oddsFormat === 'fraction'}
          value="fraction"
          onClick={e => setOddsFormat(e.target.value)}
        >
          Fraction
        </ToggleButton>
        <ToggleButton
          id="decimal"
          right
          isActive={oddsFormat === 'decimal'}
          value="decimal"
          onClick={e => setOddsFormat(e.target.value)}
        >
          Decimal
        </ToggleButton>
      </ToggleButtonWrapper>
    </ToggleWrapper>
  );
};

Toggle.propTypes = {
  oddsFormat: PropTypes.string,
  setOddsFormat: PropTypes.func
};

const mapStateToProps = state => ({
  oddsFormat: state.oddsFormat
});
const mapDispatchToProps = dispatch => ({
  setOddsFormat: value => dispatch(actions.setOddsFormat(value))
});

export default connect(mapStateToProps, mapDispatchToProps)(Toggle);
