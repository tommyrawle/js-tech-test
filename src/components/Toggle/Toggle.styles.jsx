import styled from 'styled-components';

export const ToggleWrapper = styled.div`
  color: #ffffff;
  text-align: right;
`;

export const ToggleButtonWrapper = styled.div`
  margin-top: 5px;
`;

export const ToggleButton = styled.button`
  background-color: ${({ isActive }) => (isActive ? '#e5e5e5' : '#ffffff')};
  cursor: pointer;
  &:hover {
    background-color: ${({ isActive }) => (isActive ? '#D9D9D9' : '#f2f2f2')};

  }
  border: 1px solid #e0e0e0
  text-transform: uppercase;
  color: #535353;
  border-radius: ${({ left }) => (left ? '5px 0 0 5px' : '0 5px 5px 0')};
  font-size: 16px;
  padding: 10px;
  outline: none;
`;
