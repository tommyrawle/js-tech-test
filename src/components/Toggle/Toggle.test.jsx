import React from 'react';
import { shallow } from 'enzyme';
import { Toggle } from './Toggle.jsx';

describe('Toggle component', () => {
  it('should render Toggle component', () => {
    const mockProps = {
      oddsFormat: 'fraction',
      setOddsFormat: jest.fn()
    };
    expect(shallow(<Toggle {...mockProps} />)).toMatchSnapshot();
  });

  it('fraction button correctly triggers setOddsFormat function on click', () => {
    const mockProps = {
      oddsFormat: 'fraction',
      setOddsFormat: jest.fn()
    };
    const wrapper = shallow(<Toggle {...mockProps} />);
    wrapper.find('[id="fraction"]').simulate('click', { target: { value: 'fraction' } });
    expect(mockProps.setOddsFormat).toBeCalled();
  });
  it('decimal button correctly triggers setOddsFormat function on click', () => {
    const mockProps = {
      oddsFormat: 'decimal',
      setOddsFormat: jest.fn()
    };
    const wrapper = shallow(<Toggle {...mockProps} />);
    wrapper.find('[id="decimal"]').simulate('click', { target: { value: 'decimal' } });
    expect(mockProps.setOddsFormat).toBeCalled();
  });

  it('correctly makes isActive props value truthy on click', () => {
    //   const mockProps = {
    //     oddsFormat: 'fraction',
    //     setOddsFormat: jest.fn()
    //   };
    //   const wrapper = shallow(<Toggle {...mockProps} />);
    //   wrapper.find('[value="fraction"]').simulate('click', { target: { value: 'fraction' } });
    //   expect(wrapper.prop('isActive')).toBeFalsy();
  });
});
