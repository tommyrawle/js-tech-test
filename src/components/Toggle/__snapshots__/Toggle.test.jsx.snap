// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle component should render Toggle component 1`] = `
<styled.div>
  Odds format
  <styled.div>
    <styled.button
      id="fraction"
      isActive={true}
      left={true}
      onClick={[Function]}
      value="fraction"
    >
      Fraction
    </styled.button>
    <styled.button
      id="decimal"
      isActive={false}
      onClick={[Function]}
      right={true}
      value="decimal"
    >
      Decimal
    </styled.button>
  </styled.div>
</styled.div>
`;
